### PR TITLE
refactor load command

### PR DESCRIPTION
**Pull request by DidierRLopes**
Improves load command

### CODE REVIEW

**Comment by DidierRLopes** [comment]
@Chavithra this is failing due to tests.

I've improved the `load` function, can you let me know how I should handle it from the tests perspective?

**Comment by Chavithra** [comment]
Changed a test this test:
- `tests/gamestonk_terminal/stocks/due_diligence/test_business_insider_view.py::test_price_target_from_analysts_plt`

Removed this test which I think doesn't make sense with the new code :
- `tests/gamestonk_terminal/stocks/due_diligence/test_dd_controller.py::test_call_load_no_args`

**Comment by jmaslek** [comment]
Maybe not related to this specific pr, but `ta` should have a `load`

**Comment by jmaslek** [comment]
When trying a foreign exchange:

```
(✨) (stocks)> load glxy.to
Loading Daily GLXY.TO stock with starting period 2020-12-07 for analysis.

not enough values to unpack (expected 2, got 1) 
```

File: gamestonk_terminal/stocks/stocks_helper.py [hunk 1]
```diff
@@ -140,206 +86,160 @@ def search(
...
-        Ticker
-    s_start : str
-        Start date
-    s_interval : str
-        Interval to get data for
-    df_stock : pd.DataFrame
-        Preloaded dataframe
+    ticker: str
+        Ticker to get data
+    start: str
```
Comment (jmaslek) [inline gamestonk_terminal/stocks/stocks_helper.py:126]: You have this type hinted as datetime

File: gamestonk_terminal/stocks/stocks_helper.py [hunk 2]
```diff
@@ -352,79 +252,66 @@ def load(
...
+            if df_stock_candidate.index[0] > pd.to_datetime(start):
                 s_start = df_stock_candidate.index[0]
             else:
-                s_start = ns_parser.s_start_date
+                s_start = start
 
-        elif ns_parser.source == "yf":
-            s_int = str(ns_parser.n_interval) + "m"
+        elif source == "yf":
+            s_int = str(interval) + "m"
```
Comment (jmaslek) [inline gamestonk_terminal/stocks/stocks_helper.py:277]: f string?

**Comment by jmaslek** [comment]
looks good.  works on all the menus for me

**Comment by DidierRLopes** [comment]
> looks good. works on all the menus for me

should be good now

**Comment by jmaslek** [comment]
Still getting
```
(✨) (stocks)> load glxy.to
Loading Daily GLXY.TO stock with starting period 2020-12-07 for analysis.

not enough values to unpack (expected 2, got 1) 
```

**Review by piiq — COMMENTED** [review]
It looks like a lot of code duplication is being introduces here. Each of the controllers that are touched by this PR has a load function that are really similar. If we are to change/refactor this again it would require making edits in a lot of places. Are we sure about this?

**Comment by piiq** [comment]
> That's fixed.

All international exchanges that I've tried worked as expected.

alibaba in hk
image

blackberry in toronto
image

yandex in moscow
image

File: gamestonk_terminal/stocks/stocks_helper.py [hunk 3]
```diff
@@ -26,66 +25,13 @@
...
     plot_autoscale,
-    valid_date,
+    try_except,
 )
 
-warnings.simplefilter("ignore")
 # pylint: disable=no-member,too-many-branches,C0302
 
 
-def clear(
```
[the remark this answers is not in the document]
Comment (DidierRLopes) [reply gamestonk_terminal/stocks/stocks_helper.py:36]: We aren't using this anywhere

**Comment by DidierRLopes** [comment]
> It looks like a lot of code duplication is being introduced here. Each of the controllers that are touched by this PR has a load function that are really similar. If we are to change/refactor this again it would require making edits in a lot of places. Are we sure about this?

Yup, readability >> less lines.

Also, if you look into each of these loads they allowed user to specify intraday, interval, source, ... and then what we really wanted was just either a ticker name or a start date time...

And in terms of the usage on the reports now it's much nicer than what it was. Now you don't need to send a list of strings (other_args) which is something we moved away on our main refactoring couple weeks away, and load was just waiting to be refactored similarly

**Review by jmaslek — APPROVED** [review]

**Pull request merged**